builtin/providers/aws/resource_aws_config_config_rule.go

```diff
@@ -0,0 +1,311 @@
+		ruleInput.InputParameters = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("maximum_execution_frequency"); ok {
+		ruleInput.MaximumExecutionFrequency = aws.String(v.(string))
+	}
+
+	input := configservice.PutConfigRuleInput{
+		ConfigRule: &ruleInput,
+	}
+	log.Printf("[DEBUG] Creating AWSConfig config rule: %s", input)
+	err := resource.Retry(2*time.Minute, func() *resource.RetryError {
+		_, err := conn.PutConfigRule(&input)
+		if err != nil {
+			if awsErr, ok := err.(awserr.Error); ok {
+				if awsErr.Code() == "InsufficientPermissionsException" {
+					// IAM is eventually consistent
+					return resource.RetryableError(err)
+				}
+			}
+
+			return resource.NonRetryableError(fmt.Errorf("Failed to create AWSConfig rule: %s", err))
+		}
+
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	d.SetId(name)
+
+	log.Printf("[DEBUG] AWSConfig config rule %q created", name)
+
+	return resourceAwsConfigConfigRuleRead(d, meta)
+}
+
+func resourceAwsConfigConfigRuleRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).configconn
+
+	out, err := conn.DescribeConfigRules(&configservice.DescribeConfigRulesInput{
+		ConfigRuleNames: []*string{aws.String(d.Id())},
+	})
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "NoSuchConfigRuleException" {
+			log.Printf("[WARN] Config Rule %q is gone", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	if len(out.ConfigRules) < 1 {
+		log.Printf("[WARN] Config Rule %q is gone", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	log.Printf("[DEBUG] AWS Config config rule received: %s", out)
+
+	rule := out.ConfigRules[0]
+	d.Set("arn", rule.ConfigRuleArn)
+	d.Set("rule_id", rule.ConfigRuleId)
+	d.Set("name", rule.ConfigRuleName)
+	d.Set("description", rule.Description)
+	d.Set("input_parameters", rule.InputParameters)
+	d.Set("maximum_execution_frequency", rule.MaximumExecutionFrequency)
+
+	if rule.Scope != nil {
+		d.Set("scope", flattenConfigRuleScope(rule.Scope))
+	}
+
+	var source []interface{}
+	m := make(map[string]interface{})
+	m["owner"] = *rule.Source.Owner
+	m["source_identifier"] = *rule.Source.SourceIdentifier
+	if len(rule.Source.SourceDetails) > 0 {
+		m["source_detail"] = schema.NewSet(configRuleSourceDetailsHash, flattenConfigRuleSourceDetails(rule.Source.SourceDetails))
+	}
+	source = append(source, m)
+	d.Set("source", source)
+
+	return nil
+}
+
+func resourceAwsConfigConfigRuleDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).configconn
+
+	name := d.Get("name").(string)
+
+	log.Printf("[DEBUG] Deleting AWS Config config rule %q", name)
+	_, err := conn.DeleteConfigRule(&configservice.DeleteConfigRuleInput{
+		ConfigRuleName: aws.String(name),
+	})
+	if err != nil {
+		return fmt.Errorf("Deleting Config Rule failed: %s", err)
+	}
+
+	conf := resource.StateChangeConf{
+		Pending: []string{
+			"ACTIVE",
+			"DELETING",
+			"DELETING_RESULTS",
+			"EVALUATING",
```

Comment:
Good idea, I did consider it briefly, just thought this would be shorter, but your solution is cleaner in the sense of the SDK. 👍 I will change it.